tools/xabuild/XABuild.cs

```diff
@@ -40,6 +40,14 @@ static int Main ()
 
 				//Symbolic links to be created: key=in-tree-dir, value=system-dir
 				var symbolicLinks = new Dictionary<string, string> ();
+				if (paths.IsMacOS) {
+					foreach (var dir in Directory.EnumerateDirectories (paths.MonoSystemFrameworkRoot)) {
+						if (Path.GetFileName (dir).EndsWith ("-api", StringComparison.OrdinalIgnoreCase)){
+							var inTreeFramework = Path.Combine (paths.XamarinAndroidBuildOutput, "lib", "xamarin.android", Path.GetFileName (dir));
+							symbolicLinks [inTreeFramework] = dir;
+						}
+					}
+				}
```

Comment:
Is it possible to make `SystemFrameworks` an array and add these  directories here?

https://github.com/xamarin/xamarin-android/blob/4c800554ab033b7c882a7c9812172b8c487b861d/tools/xabuild/XABuildPaths.cs#L183

Reply:
The symlink target path (in our build tree) is different for the folders in `SystemFrameworks` and `MonoSystemFrameworkRoot`. If we make this a list which contains both paths, we'd need a way to swap the symlink target path when iterating through it. I suppose we could do this instead if it's more readable:
```cs
foreach (var frameworkRoot in paths.SystemFrameworks) {
	foreach (var dir in Directory.EnumerateDirectories (frameworkRoot)) {
		if (Path.GetFileName (dir) != "MonoAndroid") {
			var inTreeFramework = Path.Combine (paths.FrameworksDirectory, Path.GetFileName (dir));
			if (paths.IsMacOS && Path.GetFileName (dir).EndsWith ("-api", StringComparison.OrdinalIgnoreCase)) {
				inTreeFramework = Path.Combine (paths.XamarinAndroidBuildOutput, "lib", "xamarin.android", Path.GetFileName (dir));
			}
			symbolicLinks [inTreeFramework]	= dir;
		}
	}
}

```

Reply:
I think what you have here is fine. Maybe we can phase out `xabuild` with .NET 5? 👍